src/main/java/com/uid2/shared/auth/ClientKey.java

```diff
@@ -48,13 +51,15 @@ public class ClientKey implements IRoleAuthorizable<Role> {
     public ClientKey(String key, String secret) {
         this.key = key;
         this.secret = secret;
+        this.secretBytes = Utils.decodeBase64String(secret);
```

Comment:
Should this call `setSecret()` to follow the DRY principle?

Reply:
done